tests/python/test_dataset_conversion.py

```diff
@@ -0,0 +1,224 @@
+    ds2.add_tensor("y",lat_2d)
+    ds2.add_meta_string("y_coord_units",lat_coord_2d[2]["y_coord_units"])
+    ds2.add_meta_string("y_coord_longname",lat_coord_2d[2]["y_coord_longname"])
+    # Add attribute data to dataset 
+    ds2.add_meta_string("units",data_attributes_1d["units"])
+    ds2.add_meta_string("longname",data_attributes_1d["longname"])
+    ds2.add_meta_string("convention",data_attributes_1d["convention"])
+    # Add dimension data to dataset 
+    ds2.add_meta_string("dim_data_x",lon_coord_2d[0])
+    ds2.add_meta_string("dim_data_y",lat_coord_2d[0])
+
+
+    # Calling add_metadata_for_xarray for 2D data
+    DatasetConverter.add_metadata_for_xarray(
+        ds2, 
+        data_names=["2ddata"],
+        dim_names=["dim_data_x","dim_data_y"],
+        coord_names=["x","y"], 
+        attr_names=["units","longname","convention"]
+    )
+    
+    # Calling add_metadata_for_xarray for longitude coordinate
+    DatasetConverter.add_metadata_for_xarray(
+    ds2,
+    data_names=["x"], 
+    dim_names=["dim_data_x"],
+    attr_names=["x_coord_units","x_coord_longname"] 
+    )
+    
+    # Calling add_metadata_for_xarray for latitude coordinate
+    DatasetConverter.add_metadata_for_xarray(
+        ds2,
+        data_names=["y"], 
+        dim_names=["dim_data_y"],
+        attr_names=["y_coord_units","y_coord_longname"]
+    )
+    
+    # Assert that created data and data put into the dataset are the same 
+    # Comparing tensor data and metadata extracted from dataset after 
+    # add_metadata_for_xarray call to generated 1D data 
+    assert(ds2.get_tensor("2ddata") == data2d).all()
+    assert(ds2.get_meta_strings("units")[0] == "m/s")
+    assert(ds2.get_meta_strings("longname")[0] =="velocity")
+    assert(ds2.get_meta_strings("convention")[0] == "CF1.5")  
+    assert(ds2.get_meta_strings("dim_data_x")[0] == "x")
+    assert(ds2.get_meta_strings("dim_data_y")[0] == "y")
+    # Comparing tensor data and metadata for coordinate data extracted from dataset after 
+    # add_metadata_for_xarray call to generated 2D coordinate data 
+    assert(ds2.get_tensor("x")==lon_2d).all()
+    assert(ds2.get_meta_strings("x_coord_units")[0]=="degrees E")    
+    assert(ds2.get_meta_strings("x_coord_longname")[0]=="Longitude")
+    assert(ds2.get_tensor("y") == lat_2d).all()
+    assert(ds2.get_meta_strings("y_coord_units")[0] == "degrees N")    
+    assert(ds2.get_meta_strings("y_coord_longname")[0] == "Latitude")
+    
+
+def test_transform_to_xarray_1d():
+ 
+# Compare generated Xarray from 1D data to initial Xarray
+    d1_xarray_ret = DatasetConverter.transform_to_xarray(ds1)
+    d1_transformed = d1_xarray_ret["1ddata"]
+    assert(ds_1d.identical(d1_transformed))
+    
+def test_transform_to_xarray_2d():
+
+# Compare generated Xarray to initial Xarray 
+    d2_xarray_ret = DatasetConverter.transform_to_xarray(ds2)
+    d2_transformed = d2_xarray_ret["2ddata"]
+    assert(ds_2d.identical(d2_transformed))
```

Comment:
This file is only checking the successful conversion paths. We need to test with bad parameters or when other things go wrong to make sure that our code will behave predictably and reasonably in these cases